packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -383,9 +383,9 @@ const DomainPicker: FunctionComponent< Props > = ( {
 																suggestionRefs.current[ index ] = ref;
 															} }
 															key={ suggestion.domain_name }
-															isUnavailable={ ! isAvailable }
 															domain={ suggestion.domain_name }
 															cost={ suggestion.cost }
+															isUnavailable={ ! isAvailable || suggestion?.unavailable }
```

Comment:
This is theoretically true, but when you true to move it up the condition becomes quite dirty. 

```js
const isAvailable =
	( availabilityStatus
		? domainIsAvailableStatus?.indexOf( availabilityStatus ) > -1
		: true ) &&
	suggestion &&
	! suggestion.unavailable;
```

Reply:
It might be a bit farfetched, but what if we declare 2 `const` and combine them?

```ts
const isDomainAvailable = availabilityStatus
		? domainIsAvailableStatus?.indexOf( availabilityStatus ) > -1
		: true;
const isSuggestionAvailable = suggestion && ! suggestion.unavailable;
const isAvailable = isDomainAvailable && isSuggestionAvailable;
```